src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (const arg of this) {
+      if (arg === null || arg === undefined) {
+        string += '';
+      } else {
+        string += arg;
+      }
```

Comment:
let's improve this part a little bit - you can leave only `if` block and check if the argument is not null **and** not undefined  

```suggestion
      if (arg !== null && arg !== undefined) {
        string += arg;
       }
```